appnav/src/main/kotlin/io/element/android/appnav/loggedin/LoggedInPresenter.kt

```diff
@@ -53,29 +56,9 @@ class LoggedInPresenter @Inject constructor(
             sessionVerificationService.sessionVerifiedStatus.map { it == SessionVerifiedStatus.Verified }
         }.collectAsState(initial = false)
 
-        LaunchedEffect(isVerified) {
-            if (isVerified) {
-                // Ensure pusher is registered
-                val currentPushProvider = pushService.getCurrentPushProvider()
-                val result = if (currentPushProvider == null) {
-                    // Register with the first available push provider
-                    val pushProvider = pushService.getAvailablePushProviders().firstOrNull() ?: return@LaunchedEffect
-                    val distributor = pushProvider.getDistributors().firstOrNull() ?: return@LaunchedEffect
-                    pushService.registerWith(matrixClient, pushProvider, distributor)
-                } else {
-                    val currentPushDistributor = currentPushProvider.getCurrentDistributor(matrixClient)
-                    if (currentPushDistributor == null) {
-                        // Register with the first available distributor
-                        val distributor = currentPushProvider.getDistributors().firstOrNull() ?: return@LaunchedEffect
-                        pushService.registerWith(matrixClient, currentPushProvider, distributor)
-                    } else {
-                        // Re-register with the current distributor
-                        pushService.registerWith(matrixClient, currentPushProvider, currentPushDistributor)
-                    }
-                }
-                result.onFailure {
-                    Timber.e(it, "Failed to register pusher")
-                }
+        if (isVerified) {
```

Comment:
Checking `isVerified` first, else `ensurePusherIsRegistered()` is called twice:

<img width="1027" alt="image" src="https://github.com/element-hq/element-x-android/assets/3940906/f50924c5-d5ee-4058-898c-0ef476dda016">

By checking first the value:

<img width="1107" alt="image" src="https://github.com/element-hq/element-x-android/assets/3940906/eeb7a089-3ce8-4ec3-8c46-cc799af8b42f">

Reply:
Fixed by [bc30aee](https://github.com/element-hq/element-x-android/pull/3035/commits/bc30aee359c834124a765fe664b548b8a765fa5d)